hydra-node/src/Hydra/API/ServerOutput.hs

```diff
@@ -81,7 +81,7 @@ data ServerOutput tx
   | -- | A friendly welcome message which tells a client something about the
     -- node. Currently used for knowing what signing key the server uses (it
     -- only knows one).
-    Greetings {me :: Party}
+    Greetings {me :: Party, headStatus :: HeadStatus, snapshotUtxo :: UTxOType tx}
```

Comment:
Should: model the utxo here as `Maybe (UTxOType tx)` and update the haddocs

Only the `Open` head status has a utxo set. Either move the utxo into the `Open` `HeadStatus` or make this an optional field. In the latter case we should check that the `ToJSON` instance does leave out the field if it is `Nothing`